modules/nw-metallb-layer2.adoc

```diff
@@ -0,0 +1,41 @@
+[id="nw-metallb-layer2_{context}"]
+= MetalLB concepts for layer 2 mode
+
+In layer 2 mode, the `speaker` pod on one node announces the external IP address for a service to the machine network.
+From a network perspective, the node appears to have multiple IP addresses assigned to a network interface.
+
+The `speaker` pod responds to ARP requests for IPv4 services and NDP requests for IPv6.
+
+In layer 2 mode, all traffic for a service IP address is routed through one node.
+After traffic enters the node, the service proxy for the CNI network provider distributes the traffic to all the pods for the service.
+
+Because all traffic for a service enters through a single node in layer 2 mode, in a strict sense, MetalLB does not implement a load balancer for layer 2.
+Rather, MetalLB implements a failover mechanism for layer 2 so that when a `speaker` pod becomes unavailable, a `speaker` pod on a different node can announce the service IP address.
+
+When a node becomes unavailable, failover is automatic.
+The `speaker` pods on the other nodes detect that a node is unavailable and a new `speaker` pod and node take ownership of the service IP address from the failed node.
+
+image::nw-metallb-layer2.png[Conceptual diagram for MetalLB and layer 2 mode]
+
+The preceding graphic shows the following concepts:
+
+* An application is available through a service that has a cluster IP of `172.30.x.x`.
+That IP address is accessible from inside the cluster.
+The service also has an external IP address that MetalLB assigned to the service, `192.168.55.22`.
+
+* Each node happens to have a pod for the application.
+However, you are not required to run an instance of the application on every node.
+
+* The `speaker` daemon set runs a pod on each node.
+The MetalLB Operator starts these pods.
+
+* Each `speaker` pod is a host-networked pod.
+The IP address for the pod is identical to the IP address for the node on the machine network.
+
+* The `speaker` pod on node 1 uses ARP or NDP to announce the external IP address for the service, `192.168.55.22`.
```

Comment:
the example used here for IPv4 so no NDP correct ?

Reply:
Yes and thank you.